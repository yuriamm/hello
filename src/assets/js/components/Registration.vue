<template>
  <div>
    <NavBar :isLoggedIn="isLoggedIn" />
    <div v-if="isLoggedIn">You're Logged In!</div>
    <div v-else>
      <h1>Register</h1>
      <div class="alert alert-danger" v-if="error">Error!</div>
      <form class="register" @submit.prevent="register">
        <input
          v-model="user.username"
          placeholder="username"
          type="text"
          required
        />
        <input
          v-model="user.password"
          placeholder="password"
          type="password"
          required
        />
        <input
          v-model="user.password_confirmation"
          placeholder="password confirmation"
          type="password"
          required
        />

        <button type="submit">Submit</button>
      </form>
    </div>
  </div>
</template>

<script>
import request from "../api/request.js";
import NavBar from "./NavBar";
export default {
  name: "registration",
  components: {
    NavBar
  },
  data() {
    return {
      user: {
        username: "",
        password: "",
        password_confirmation: ""
      },
      error: false,
      isLoggedIn: false
    };
  },
  methods: {
    async register() {
      const response = await request("/api/signup", this.user, "post");
      if (response.data.message === "success") {
        sessionStorage.setItem("user_id", response.data.user);
        this.isLoggedIn = true;
        window.location.href = "/home";
      }
      if (response.data.message === "error") {
        this.error = true;
      }
    }
  }
};
</script>
