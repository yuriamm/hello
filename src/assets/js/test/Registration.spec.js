import { shallowMount } from "@vue/test-utils";
import Registration from "../components/Registration.vue";
jest.mock("../api/request.js");

global.window = Object.create(window);
const url = "http://localhost:4000";
Object.defineProperty(window, "location", {
  value: {
    href: url,
  },
});

const wrapper = shallowMount(Registration);

describe("Session.vue", () => {
  it("snapshot matches", async () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("snapshot matches when logged in", async () => {
    await wrapper.setData({ isLoggedIn: true });

    expect(wrapper.element).toMatchSnapshot();
  });

  it("redirects to correct path", async () => {
    await wrapper.setData({
      isLoggedIn: false,
      user: {
        username: "username",
        password: "password",
        password_confirmation: "password",
      },
    });
    wrapper.find("form").trigger("submit.prevent");
    wrapper.vm.$nextTick(() => {
      expect(window.location.href).toEqual("/home");
      done();
    });
  });
});
