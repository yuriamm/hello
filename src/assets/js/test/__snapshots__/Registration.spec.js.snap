// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Session.vue snapshot matches 1`] = `
<div>
  <navbar-stub />
   
  <div>
    <h1>
      Register
    </h1>
     
    <!---->
     
    <form
      class="register"
    >
      <input
        placeholder="username"
        required="required"
        type="text"
      />
       
      <input
        placeholder="password"
        required="required"
        type="password"
      />
       
      <input
        placeholder="password confirmation"
        required="required"
        type="password"
      />
       
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;

exports[`Session.vue snapshot matches when logged in 1`] = `
<div>
  <navbar-stub
    isloggedin="true"
  />
   
  <div>
    You're Logged In!
  </div>
</div>
`;
